apps/arweave/src/ar_chain_stats.erl

```diff
@@ -0,0 +1,176 @@
+-module(ar_chain_stats).
+
+-behaviour(gen_server).
+
+-include_lib("arweave/include/ar.hrl").
+-include_lib("arweave/include/ar_chain_stats.hrl").
+-include_lib("eunit/include/eunit.hrl").
+
+-export([log_fork/2, get_forks/1]).
+
+-export([start_link/0, init/1, handle_call/3, handle_cast/2, handle_info/2, terminate/2]).
+
+%%%===================================================================
+%%% Public interface.
+%%%===================================================================
+start_link() ->
+    gen_server:start_link({local, ?MODULE}, ?MODULE, [], []).
+
+log_fork([], _ForkRootB) ->
+    %% No fork to log
+    ok;
+log_fork(Orphans, ForkRootB) ->
+    gen_server:cast(?MODULE, {log_fork, Orphans, ForkRootB, os:system_time(millisecond)}).
+
+%% @doc Returns all forks that have been logged since the given start time
+%% (system time in seconds)
+get_forks(StartTime) ->
+    gen_server:call(?MODULE, {get_forks, StartTime}).
```

Comment:
I'd handle the timeout gracefully here, I believe, we'll hit it every once in a while.

Reply:
Updated: https://github.com/ArweaveTeam/arweave/pull/581/commits/b63151dc1abd41661302bb5fce2f810f7a5cd7dc